blackjax/inference/metrics.py

```diff
@@ -95,7 +95,10 @@ def gaussian_euclidean(
         )
 
     def momentum_generator(rng_key: jnp.ndarray, position: PyTree) -> PyTree:
-        _, unravel_fn = ravel_pytree(position)
+        ravelled_position, unravel_fn = ravel_pytree(position)
+        assert ravelled_position.shape[0] == shape[0], \
+            f"'inverse_mass_matrix' and 'position' are not compatible. 'inverse_mass_matrix' has dimension {shape}, " \
+            f"but 'position' ravelled dimension is {ravelled_position.shape[0]}."
```

Comment:
this would be skipped in optimised mode, perhaps just check if they're the same and raise an error if not?

Reply:
What do you mean skipped in optimized mode? Also I that what I am doing is exactly checking that they are the same and raising an error otherwise no?